src/kudu/consensus/raft_consensus.cc

```diff
@@ -4427,7 +4450,26 @@ void RaftConsensus::HandleProxyRequest(const ConsensusRequestPB* request,
 
 Status RaftConsensus::SetCompressionCodec(const std::string& codec) {
   LockGuard l(lock_);
-  return queue_->log_cache()->SetCompressionCodec(codec);
+  if (codec.empty()) {
+    LOG(INFO) << "Disabling compression";
+    codec_ = nullptr;
+    return Status::OK();
```

Comment:
this is only setting the local value of codec_. We are not resetting the value in LogCache.